config/initializers/preload_stis.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Taken from https://github.com/rails/rails/blob/main/guides/source/autoloading_and_reloading_constants.md
+# Makes ParticipantProfile::ECF behave properly in dev and test
+unless Rails.application.config.eager_load
+  Rails.autoloaders.main.on_load("ParticipantProfile") do
+    ParticipantProfile.connection.select_values(<<~SQL).each(&:constantize)
```

Comment:
I just love how github tells me "Possible SQL injection". 

But we control everything that goes into this query, right?

Reply:
(I guess we could only enable it in dev and test if we really needed to)

Reply:
Yes, you'd have to inject something into the database in the type column to do anything malicious I think. This will already only run on dev/test because of the `unless Rails.application.config.eager_load`